### PR TITLE
Moved Currency stuff in ContractsDSL out of core and into finance

### PR DESCRIPTION
**Pull request by shamsasari**

### CODE REVIEW

File: client/jfx/src/integration-test/kotlin/net/corda/client/jfx/NodeMonitorModelTest.kt [hunk 1]
```diff
@@ -5,8 +5,8 @@ import net.corda.client.jfx.model.ProgressTrackingEvent
 import net.corda.core.internal.bufferUntilSubscribed
 import net.corda.core.contracts.Amount
```
Comment (mnesbit) [inline client/jfx/src/integration-test/kotlin/net/corda/client/jfx/NodeMonitorModelTest.kt]: Can we try to drop the module dependency of :client:jfx on :finance down to testCompile?

**Comment by mnesbit** [comment]
Last time I tried this the web stuff messed me up. It will need retesting manually.

File: client/jfx/src/integration-test/kotlin/net/corda/client/jfx/NodeMonitorModelTest.kt (hunk 1 above)
Comment (shamsasari) [inline client/jfx/src/integration-test/kotlin/net/corda/client/jfx/NodeMonitorModelTest.kt]: It's needed as a compile dependency, e.g. at least `ContractStateModel` needs it.

**Comment by shamsasari** [comment]
Demos look OK.

**Pull request merged**